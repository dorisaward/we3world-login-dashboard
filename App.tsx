import React, {useState, useCallback} from 'react';
import {
  Button,
  SafeAreaView,
  ScrollView,
  StyleProp,
  Text,
  View,
  ViewStyle,
} from 'react-native';
import {LoginScreen} from './src/screens/login/LoginScreen';

function App(): React.JSX.Element {
  const [screenToDisplay, setScreenToDisplay] = useState<screen>(screen.Login);
  const navigateToDashboard = useCallback(
    () => setScreenToDisplay(screen.Dashboard),
    [],
  );
  const navigateToLoginScreen = useCallback(
    () => setScreenToDisplay(screen.Login),
    []
  )
  const backgroundStyle: StyleProp<ViewStyle> = {
    backgroundColor: '#f5f5dc',
    height: '100%',
  };

  return (
    <SafeAreaView style={backgroundStyle}>
      <ScrollView contentInsetAdjustmentBehavior="automatic">
        <View>
          <Text>We3World Login Page</Text>
          {screenToDisplay === screen.Login ? (
            <LoginScreen navigateToDashboard={navigateToDashboard} />
          ) : <Button title='logout' onPress={navigateToLoginScreen}/>}
        </View>
      </ScrollView>
    </SafeAreaView>
  );
}

enum screen {
  Login = 'Login',
  Dashboard = 'Dashboard',
}

export default App;
