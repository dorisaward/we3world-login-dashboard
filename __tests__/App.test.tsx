import 'react-native';
import React from 'react';
import App from '../App';
import {it, expect} from '@jest/globals';
import {render} from '@testing-library/react-native';

jest.mock('../src/screens/login/LoginScreen');

it('renders correctly', () => {
  // Given
  const renderable = <App />;

  // When
  const {toJSON} = render(renderable);

  // Then
  expect(toJSON()).toMatchSnapshot();
});
