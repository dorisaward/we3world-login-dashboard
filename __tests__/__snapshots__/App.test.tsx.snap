// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#f5f5dc",
      "height": "100%",
    }
  }
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
  >
    <View>
      <View>
        <Text>
          We3World Login Page
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
